packages/rest/src/__tests__/acceptance/routing/routing.acceptance.ts

```diff
@@ -361,30 +356,80 @@ describe('Routing', () => {
       });
   });
 
-  it('binds the current controller', async () => {
-    const app = givenAnApplication();
-    const server = await givenAServer(app);
-    const spec = anOpenApiSpec()
-      .withOperationReturningString('get', '/name', 'checkController')
-      .build();
+  describe('current controller', () => {
+    let app: Application;
+    let server: RestServer;
+    let controllerClass: Constructor<ControllerInstance>;
 
-    @api(spec)
-    class GetCurrentController {
-      async checkController(
-        @inject('controllers.GetCurrentController') inst: GetCurrentController,
-      ): Promise<object> {
-        return {
-          result: this === inst,
-        };
+    beforeEach(setupController);
+
+    it('binds current controller resolved from a transient binding', async () => {
+      givenControllerInApp(app, controllerClass);
+
+      await whenIMakeRequestTo(server)
+        .get('/name')
+        .expect({
+          count: 1,
+          isSingleton: false,
+          result: true,
+        });
+
+      // Make a second call
+      await whenIMakeRequestTo(server)
+        .get('/name')
+        .expect({
+          count: 1, // The count is still 1 as it's from a new instance
+          isSingleton: false,
+          result: true,
+        });
+    });
+
+    it('binds current controller resolved from a singleton binding', async () => {
+      givenControllerAsSingletonInApp(app, controllerClass);
```

Comment:
When reading the code for the first time, I find the overhead of reading yet another helper implementation as too much. Have you considered the following?

```suggestion
      givenControllerInApp(app, controllerClass).inScope(BindingScope.SINGLETON)
```

Considering how little is `givenControllerInApp` doing, I would get rid of it entirely.

```suggestion
      app.controller(controllerClass).inScope(BindingScope.SINGLETON)
```

Reply:
> Considering how little is `givenControllerInApp` doing, I would get rid of it entirely.

☝️